File: root/test_post.py
#!/usr/bin/env python3

import os
import sys

print("Content-Type: text/html")
print("Status: 200 OK")
print()  # Empty line separates headers from body

print("<!DOCTYPE html>")
print("<html>")
print("<head><title>Python CGI POST Test</title></head>")
print("<body>")
print("<h1>Hello from Python CGI!</h1>")

# Check if this is a POST request
request_method = os.environ.get('REQUEST_METHOD', '')
print(f"<p>Request Method: {request_method}</p>")

if request_method == 'POST':
    # Read POST data from stdin
    content_length = os.environ.get('CONTENT_LENGTH', '0')
    if content_length and content_length.isdigit():
        post_data = sys.stdin.read(int(content_length))
        print(f"<p>POST Data Received: {post_data}</p>")
        
        # Parse the form data
        if post_data:
            pairs = post_data.split('&')
            print("<ul>")
            for pair in pairs:
                if '=' in pair:
                    key, value = pair.split('=', 1)
                    print(f"<li>{key} = {value}</li>")
            print("</ul>")
    else:
        print("<p>No POST data found</p>")
else:
    print("<p>This was not a POST request</p>")

print("</body>")
print("</html>")

File: www/cgi/php/test_post.php
#!/usr/bin/php-cgi
<?php
echo "Content-Type: text/html\r\n\r\n";
echo "<h3>Environment Test</h3>";
echo "<p>REQUEST_METHOD: " . getenv('REQUEST_METHOD') . "</p>";
echo "<p>CONTENT_LENGTH: " . getenv('CONTENT_LENGTH') . "</p>";
echo "<p>CONTENT_TYPE: " . getenv('CONTENT_TYPE') . "</p>";

$stdin = fopen('php://stdin', 'r');
$data = '';
$len = intval(getenv('CONTENT_LENGTH'));
if ($len > 0) {
    $data = fread($stdin, $len);
}
fclose($stdin);

echo "<p>RAW POST Data: [" . htmlspecialchars($data) . "]</p>";
echo "<p>Data length: " . strlen($data) . "</p>";
?>
